### DISCUSSION TITLE
Difficulty making a 3 level chained select

### DISCUSSION THREAD

**Discussion by djkunkel** [vueform/vueform#95] [Questions] [answered]
I'm attempting to make a 3 level chained select and it seems like it refuses to use the second variable in my endpoint for the third and final select. It puts the urlencoded variable statement (/regions/%7Bstate%7C'none'%7D/index.json) into the url instead of replacing it with the value from the previous select. The other variable in the url is replaced correctly.

I've setup a simple demo to demonstrate:
https://stackblitz.com/edit/vitejs-vite-nt23dy?file=App.vue

Basically I'm trying to have a country selector, then a state/province selector, and finally a region selector.

I also noticed that it makes requests even if the previous selector has not yet been selected, is there a way to prevent a select from calling its endpoint until the variable is available? I tried setting the :disabled property to true if the parent select doesn't have a value, but it seems to call the endpoint anyway.

Thank you!

**Comment by adamberecz** [comment, marked as answer]
Thanks for reporting the issue @djkunkel. Fixed in `1.6.5`.

Also, you can disable loading options until the variables are not set like this:

```vue
<Vueform>
  <SelectElement
    name="country"
    :search="true"
    :native="false"
    :label="{
      en: 'Country',
    }"
    input-type="search"
    autocomplete="off"
    items="/countries/index.json"
  />
  <SelectElement
    name="state"
    :search="true"
    :native="false"
    input-type="search"
    autocomplete="off"
    :label="{
      en: 'State/Province',
    }"
    items="/countries/{country|'none'}/index.json"
    :resolve-on-load="false"
    @open="(el$) => {
      if (!el$.input.extendedOptions.length && el$.form$.el$('country').value) {
        el$.updateItems()
      }
    }"
  />
  <SelectElement
    name="region"
    :search="true"
    :native="false"
    :label="{
      en: 'County/Region',
    }"
    input-type="search"
    autocomplete="off"
    items="/countries/{country|'none'}/regions/{state|'none'}/index.json"
    :resolve-on-load="false"
    @open="(el$) => {
      if (!el$.input.extendedOptions.length && el$.form$.el$('state').value && el$.form$.el$('country').value) {
        el$.updateItems()
      }
    }"
  />
</Vueform>
```

The `:resolve-on-load` prevents loading the options initially and you can load them on `@open` if they haven't been loaded yet and the sources for variables have values.

**Reply by djkunkel** [reply]
Thank you for this fast response. I can confirm this fixed my issue.
